clients/unshaded/src/main/java/tachyon/client/file/FileOutStream.java

```diff
@@ -136,6 +150,22 @@ public void close() throws IOException {
       }
     }
 
+    canComplete = storeToTachyon();
```

Comment:
Shouldn't `canComplete` still be true if the under storage write succeeded despite the tachyon write failing?